pkg/core/annotation.go

```diff
@@ -49,6 +49,22 @@ func (a *Annotation) MarshalJSON() ([]byte, error) {
 	return json.Marshal(m)
 }
 
+func DedupeAnnotationKeys(a []AnnotationKey) []AnnotationKey {
```

Comment:
you can speed this up a fair amount if the list gets large with a map:

`func DedupeAnnotationKeys(a []AnnotationKey) []AnnotationKey {
	result := make(map[AnnotationKey]struct{})
	for _, key := range a {
		result[key] = struct{}{}
	}
	keys := make([]AnnotationKey, 0, len(result))
	for key := range result {
		keys = append(keys, key)
	}
	return keys
}`

Reply:
makes sense, can do that, just since we really only ever have like 1 or 2 in the list that we are watching for dedupes i wasnt too worried about speed, but might as well optimize

Reply:
I mention this somewhere else I think, but might be worth making the annotation keys a set, that way we don't have to remember when to dedupe or not.